### ISSUE TITLE
xreplace on `Mul` switches order and breaks equality for real_double

### ISSUE THREAD

**Issue by bocklund** [symengine/symengine#1899]
I wrote up a simple test case that demonstrates that `xreplace` switches the arguments of a mul, which breaks equality.

```cpp
TEST_CASE("xreplace equality")
{
    RCP<const Basic> x = symbol("x");
    map_basic_basic d;
    d.clear();
    RCP<const Basic> r1 = mul(real_double(2.0), x);
    // xreplace with an empty mapping dict should be a no-op
    // r1 should be equal to r2
    RCP<const Basic> r2 = r1->xreplace(d);
    std::cout << "r1, " << *r1 << std::endl;
    std::cout << "r2, " << *r2 << std::endl;

    REQUIRE(eq(*r1, *r2));  // fails
}
```

The output is:

```
r1, 2.0*x
r2, x*2.0
```

I'm not sure whether this is an issue with how equality is defined or an issue with `xreplace` breaking an assumption about the ordering of the arguments. My understanding is that `mul` has a canonical ordering for, such that `mul(real_double(2.0), x)` and `mul(x, real_double(2.0))` both give `2.0*x`, so I think this is the latter.

As an additional note, the test passes (and doesn't switch the order) for `integer(2)`, but fails for `real_double(2.0)`.

**Comment by isuruf** [comment]
Can you send the test as a PR?

**Comment by isuruf** [comment]
Fixed in #1900
